src/app.js

```diff
@@ -136,16 +136,18 @@ const App = Marionette.Application.extend({
 
     this.setRegion(opts.region);
 
-    this.triggerMethod('before:start', options);
-
     opts.state = this.getInitState(opts.state);
 
-    this._isRunning = true;
-
     this.initState(opts);
 
+    this.triggerMethod('before:start', options);
+
+    this._isRunning = true;
+
     this.triggerStart(opts);
 
+    this.delegateStateEvents();
```

Comment:
I think I agree. In my experience, the main problem we need to solve is making the state model available to `onBeforeStart`. I'm not sure it's necessary to delay state events that long. I think calling `delegateStateEvents` after `_isRunning = true` is sufficient.